package.json

```diff
@@ -5,8 +5,8 @@
   "description": "Fastish HTML",
   "main": "index.js",
   "engines": {
-    "node": "^8.0.0",
```

Comment:
I thought this config itself would enforce the node version? Is it not the case in the AMP env here?

Reply:
Similar feedback here.

Since there is a `preinstall` script, it should run before installing via either `npm install` or `yarn install`. If version checks are done by the AMP specific script is an engine field needed at all?

Reply:
Fair points.

The thinking used to be that the `preinstall` script would merely warn you if you had a *slightly* old version, but allow the install to proceed, while the `engines` section would error out if a *really* old version was detected.

Now that the two checks have converged, what if I remove the `engines` section and print errors instead of warnings in the `preinstall` script?

Reply:
@lannka @kristoferbaxter New commit posted. PTAL.

Reply:
Good point @rsimha. Perhaps the warning and error could be controlled in the singular place (node script executed in preinstall) versus spread in engines and scripts.

Engines doesn't have a flag for "lts major version", but that is enforceable in the script executed in preinstall.

Reply:
@kristoferbaxter Correct, `engines` doesn't support specifying `lts` as a version. See https://github.com/ampproject/amphtml/pull/14029#issue-175371645

Reply:
Also, I still think we should keep the hard requirements in `engines` since it natively gives us the semantic version checks for `^8.11.1` and `^1.6.0` for free. Trying to implement these checks in the `preinstall` script without using a library like `semver` is error prone. And we can't use `semver` or any other library in the script because it will break when the script is run during first-time setup (when `node_modules` is empty).